src/main/java/org/adridadou/ethereum/EthereumProxy.java

```diff
@@ -184,7 +184,9 @@ public EthValue getBalance(EthAddress address) {
         return ethereum.getBalance(address);
     }
 
-    private void increasePendingTransactionCounter(EthAddress address) {
-        pendingTransactions.put(address, pendingTransactions.getOrDefault(address, BigInteger.ZERO).add(BigInteger.ONE));
+    private void increasePendingTransactionCounter(EthAddress address, EthHash hash) {
+        Set<EthHash> hashes = pendingTransactions.getOrDefault(address, new HashSet<EthHash>());
```

Comment:
This is totally correct but I need to rant. I don't udnerstand why the Java API cannot have a lambda in this case !? I find it stupid that you need to recreate a Set everytime. 

Just to be safe, I think that using Collections.synchronizedSet(Set set) to avoid having issues would be a good idea. 

Also, to avoid concurrent issues, I would change this code to:

Set<EthHash> hashes = pendingTransactions.computeifAbsent(address, (address) -> Collections.synchronizedSet(new HashSet<>()));

//do what you have to do


This way you avoid overriding the hashSet